### ISSUE TITLE
Kick reason

### ISSUE THREAD

**Issue by Theoneflop** [Anuken/Mindustry-Suggestions#3018]
### Describe the content or mechanics you are proposing.

Hi!
So today I was kicked from a russian server while I was just upgrading defenses and doing nothing wrong 

• (for retards that will say i probably did something bad: no i did not consume resources and no i did not bother anyone) 

So my point is, Let Mindustry record the chat that happens 7 seconds before someone is kicked until they get kicked. And when we try to rejoin the same server, mindustry should show that log.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

Well this will basically help us, people that do not speak the same language as the others on foreign servers, to be able to know later on why we got kicked (using google translate) and to report people that kick us for no reason.
This would also help wrongly-kicked people that did not notice that someone initiated a vote against them.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by SMOLKEYS** [comment]
Interesting.

**Comment by begginerr** [comment]
yes, i think they should force a kick reason too because i remember joining a server with 2 other people and they kicked me for 0 reason

**Comment by itcannotbe** [comment]
Sound like something a server plugin would solve.

**Comment by itcannotbe** [comment]
Also why are you playing on a russian server without the ability to understand russian in the first place?

**Comment by begginerr** [comment]
> Also why are you playing on a russian server without the ability to understand russian in the first place?

good question

**Comment by Theoneflop** [comment]
> Sound like something a server plugin would solve.

Not really. A plugin being available does not mean that servers will push to implement it. If I wanted to suggest a "plugin", I wouldn't be here suggesting this to Anuken.

**Comment by Theoneflop** [comment]
> Also why are you playing on a russian server without the ability to understand russian in the first place?

While your question does not contribute anything to the suggestion and is rather personal and normally I would not answer it, Here is the answer: 
1. In the time that I play, most servers have zero players while the russian server has players.
2. The server being Russian does not mean I shouldn't or I can't play on it as MOST features do not require me to understand their language as I just build,destroy and vote. The only major drawback is Voting Reason which is why I'm recommending it here for better implementation of the community to be more inclusive rather than language based.

Hope you got your answer.

**Comment by Theoneflop** [comment]
> > Also why are you playing on a russian server without the ability to understand russian in the first place?
> 
> good question

In what way?

**Comment by itcannotbe** [comment]
Take your beef with the owners/moderator of the russian mindustry server being referred.

A kick reason upon kicking or reconnecting after kicked is available for them to use.

**Comment by Theoneflop** [comment]
> Take your beef with the owners/moderator of the russian mindustry server being referred.
> 
> A kick reason upon kicking or reconnecting after kicked is available for them to use.

Well I made a suggestion, If anuken wants it, he takes it.
